sympy/crypto/crypto.py

```diff
@@ -3324,3 +3325,138 @@ def decipher_bg(message, key):
         orig_msg += (m ^ b)
 
     return orig_msg
+
+################# Columnar Transposition Cipher #################
+
+def encipher_columnar(message, key):
+    """
+    Performs Regular Columnar Encryption on plaintext and returns ciphertext
+
+    Examples
+    ========
+
+    >> from sympy.crypto.crypto import encipher_columnar
+    >> message = "HELLO WORLD"
+    >> encipher_columnar(message,"SYM")
+    "LWLEHLODEORY"
+
+    Parameters
+    ==========
+
+    message : string, the message to be encrypted.
+    key : string, the key to be used for encryption.
+
+    Returns
+    =======
+
+    The Encrypted message.
+
+    References
+    ==========
+    https://en.wikipedia.org/wiki/Transposition_cipher#Columnar_transposition
+
+    """
+    message =  ''.join(alphabet for alphabet in message if alphabet.isalpha())
+    key =  ''.join(alphabet for alphabet in key if alphabet.isalpha())
+    row_length = len(key)
+    words = len(message)
+    col_length = -(-words//row_length)          # Ceil Function
+    remainder = words%row_length
+
+    for i in range(row_length - remainder):
+        message = message + str(chr(randint(0,25)+97))      # Padding message for regular columnar transposition
+
+    rows = [ord(word) for word in key.lower()]
```

Comment:
The following will calculate the `index_list` and handle repeats:
```python
d = defaultdict(list)
low = key.lower()
for i,c in enumerate(low):
    d[c].append(i)
index_list = [j for i,j in sorted(zip([d[c].pop(0) for c in sorted(low)], range(row_length)))]
```
Note: if you use this you won't have to add 1 in `index_list.index(i + 1)`.